sklearn/tests/metadata_routing_common.py

```diff
@@ -25,55 +26,69 @@
 from sklearn.utils.multiclass import _check_partial_fit_first_call
 
 
-def record_metadata(obj, method, record_default=True, **kwargs):
-    """Utility function to store passed metadata to a method.
+def record_metadata(obj, record_default=True, **kwargs):
+    """Utility function to store passed metadata to a method of obj.
 
     If record_default is False, kwargs whose values are "default" are skipped.
     This is so that checks on keyword arguments whose default was not changed
     are skipped.
 
     """
+    stack = inspect.stack()
+    method = stack[1].function
+    parent = stack[2].function
     if not hasattr(obj, "_records"):
         obj._records = {}
+    if method not in obj._records:
+        obj._records[method] = {}
+    if parent not in obj._records[method]:
+        obj._records[method][parent] = []
     if not record_default:
         kwargs = {
             key: val
             for key, val in kwargs.items()
             if not isinstance(val, str) or (val != "default")
         }
-    obj._records[method] = kwargs
+    obj._records[method][parent].append(kwargs)
 
 
-def check_recorded_metadata(obj, method, split_params=tuple(), **kwargs):
+def check_recorded_metadata(obj, method, parent, split_params=tuple(), **kwargs):
     """Check whether the expected metadata is passed to the object's method.
 
     Parameters
     ----------
     obj : estimator object
         sub-estimator to check routed params for
     method : str
-        sub-estimator's method where metadata is routed to
+        sub-estimator's method where metadata is routed to, or the 'callee'
```

Comment:
For the diff `", or the 'callee'` do you mean that we also refer this method as "the callee"?

If this is case, I think that we should slightly reformutaled.


```suggestion
        sub-estimator's method where metadata is routed to that we refer
        it as the "callee"
```